docs/developer-docs/smart-contracts/advanced-features/handling-get-post-requests.mdx

```diff
@@ -183,59 +183,178 @@ Check out the [Azle documentation](https://github.com/demergent-labs/azle/tree/m
 <TabItem value="python" label="Python" default>
 
 ```python
-from kybra import blob, Func, nat16, Opt, query, Query, Record, Tuple, Variant, Vec
+from kybra import (
+    Alias,
+    blob,
+    Func,
+    ic,
+    nat,
+    nat16,
+    Opt,
+    Query,
+    query,
+    Record,
+    StableBTreeMap,
+    Tuple,
+    update,
+    Variant,
+    Vec,
+)
 
 
-class HttpRequest(Record):
-    method: str
-    url: str
-    headers: Vec["Header"]
-    body: blob
+class Token(Record):
+    arbitrary_data: str
 
 
-class HttpResponse(Record):
-    status_code: nat16
-    headers: Vec["Header"]
+class StreamingCallbackHttpResponse(Record):
     body: blob
-    streaming_strategy: Opt["StreamingStrategy"]
-    upgrade: Opt[bool]
+    token: Opt[Token]
 
 
-Header = Tuple[str, str]
+Callback = Func(Query[[Token], StreamingCallbackHttpResponse])
 
 
-class StreamingStrategy(Variant):
-    Callback: "CallbackStrategy"
+class CallbackStrategy(Record):
+    callback: Callback
+    token: Token
 
 
-class CallbackStrategy(Record):
-    callback: "Callback"
-    token: "Token"
+class StreamingStrategy(Variant, total=False):
+    Callback: CallbackStrategy
 
 
-Callback = Func(Query[["Token"], "StreamingCallbackHttpResponse"])
+HeaderField = Alias[Tuple[str, str]]
 
 
-class StreamingCallbackHttpResponse(Record):
+class HttpResponse(Record):
+    status_code: nat16
+    headers: Vec[HeaderField]
     body: blob
-    token: Opt["Token"]
+    streaming_strategy: Opt[StreamingStrategy]
+    upgrade: Opt[bool]
 
 
-class Token(Record):
-    arbitrary_data: str
+class HttpRequest(Record):
+    method: str
+    url: str
+    headers: Vec[HeaderField]
+    body: blob
```

Comment:
Might be missing a field here about certificates...https://internetcomputer.org/docs/current/references/http-gateway-protocol-spec#http-request-encoding

Reply:
Added certificate field